File: src/main/java/br/anhembi/funmodechild/entity/Payment.java
package br.anhembi.funmodechild.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DocumentReference;

import java.time.LocalDateTime;

@Data
@Document("payments")
public class Payment {

    @Id
    private long id;
    @DocumentReference(lazy = true)
    private Order order;
    private String cardNumber;
    private String cardName;
    private int month;
    private int year;
    private int cvv;
    private int statements;
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: src/main/kotlin/br/anhembi/funmodechild/entity/Payment.kt
package br.anhembi.funmodechild.entity

import br.anhembi.funmodechild.model.response.PaymentResponse
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.DocumentReference
import java.time.LocalDateTime

@Document("payments")
data class Payment(
    @Id
    val id: String? = null,
    @DocumentReference(lazy = true)
    var order: Order? = null,
    val cardNumber: String,
    val cardName: String,
    val month: Int,
    val year: Int,
    val cvv: Int,
    val statements: Int,
    val createdAt: LocalDateTime = LocalDateTime.now()
) {
    fun toApiResponse() = PaymentResponse(
        id = this.id!!,
        createdAt = this.createdAt,
        statements = this.statements,
        totalPrice = this.order!!.totalPrice
    )
}
